skills/timezone.js

```diff
@@ -1,63 +1,97 @@
-/*
+const {findTimeZone, getZonedTime} = require('timezone-support');
+const {formatZonedTime} = require('timezone-support/dist/parse-format');
+const ExpressionList = require('../submodules/Regex/ExpressionList');
+const timeZoneExList = require('../submodules/Regex/timeZoneExList');
 
- This simply hears the time in United States timezones and reports the system time (EST) and adjusts it accordingly
-
-*/
+// Zone Name comes from the npm module timezone-support and pulled from a list listTimeZones.
+// Using zone names will account for Daylights Savings Time
+const zones = {
+  PST: 'America/Los_Angeles',
+  MST: 'America/Phoenix',
+  CST: 'America/Chicago',
+  EST: 'America/New_York',
+  IST: 'Indian/Mahe',
+  GMT: 'Europe/London',
+  CEST: 'Europe/Warsaw',
+  CTC: 'Asia/Shanghai'
+};
 
-module.exports = function(controller) {
-  //listening for time
+module.exports = controller => {
   controller.hears(
-    ['^time (.*)'],
-    'direct_message,direct_mention',
-    (bot, message) => {
-      const timezone = message.match[1];
-      const time = getTimeForTimezone(timezone);
-      bot.createConversation(message, (err, convo) => {
-        if (err) return false;
-        const responseMessage = 'The current time is ' + time + '.';
-        convo.say(responseMessage);
-        convo.activate();
-      });
-    }
-  );
+      [ExpressionList.timezone], 
+      'direct_message,direct_mention', 
+      async (bot, message) => {
+          var tz = message.text.split(' ')[1];
+          var responseMessage = "It is "; // Prep response
 
-  //get system time
-  const getTimeForTimezone = function(timezone) {
-    var date = new Date();
-    let hours = date.getHours();
-    let adjustedHours = hours + adjustForTimezone(timezone);
-    let minutes = (date.getMinutes() < 10 ? '0' : '') + date.getMinutes();
-    return adjustedHours + ':' + minutes;
-  };
+          bot.createConversation(message, async(err, convo) => {
+              if (err) return false;
+              if(tz == undefined){
+                responseMessage = ("Missing parameter. Please provide a timezone. Use HELP for a list of timezones");
+              }else{
+                // Cannot use tz to key the expression list or zones dict becuase the user input
+                // could be a value that is not the key. Ex: Using et instead of est
+                // If we don't car about other ways to write certain timzones we can simplify
+                switch(true){
```

Comment:
Can this giant switch:case block be refactored into a get function? It's a good habit to strip out single purpose chunks of code into smaller functions, example:

```javascript
const getResolveTimezone = (timezone) => {
  switch(true) {
    case(something.test(timezone)):
       return adjustForTimezone(zones.PST);
    case(....):
       return adjustFor....;
  }
}
```